packages/ckeditor5-ui/tests/dialog/dialog.js

```diff
@@ -0,0 +1,368 @@
+			} )
+			.then( newEditor => {
+				editor = newEditor;
+				dialogPlugin = editor.plugins.get( Dialog );
+			} );
+	} );
+
+	afterEach( () => {
+		editor.destroy();
+		editorElement.remove();
+		Dialog.visibleDialogPlugin = undefined;
+	} );
+
+	it( 'should initialise without #visibleDialogPlugin set', () => {
+		expect( Dialog.visibleDialogPlugin ).to.be.undefined;
+	} );
+
+	describe( 'constructor()', () => {
+		describe( 'should initialise', () => {
+			describe( '`show` event listeners', () => {
+				it( 'executing `_show()` method', () => {
+					const spy = sinon.spy( dialogPlugin, '_show' );
+
+					dialogPlugin.fire( 'show', {} );
+
+					sinon.assert.calledOnce( spy );
+				} );
+
+				it( 'executing `onShow` callback', () => {
+					const spy = sinon.spy( () => {} );
+
+					dialogPlugin.fire( 'show', { onShow: spy } );
+
+					sinon.assert.calledOnce( spy );
+				} );
+
+				it( 'should execute `_show()` before `onShow`', () => {
+					const _showSpy = sinon.spy( dialogPlugin, '_show' );
+					const onShowSpy = sinon.spy( () => {} );
+
+					dialogPlugin.on( 'show', () => {
+						sinon.assert.notCalled( _showSpy );
+						sinon.assert.notCalled( onShowSpy );
+					}, { priority: 'high' } );
+
+					dialogPlugin.on( 'show', () => {
+						sinon.assert.calledOnce( _showSpy );
+						sinon.assert.notCalled( onShowSpy );
+					}, { priority: -1 } );
+
+					dialogPlugin.on( 'show', () => {
+						sinon.assert.calledOnce( _showSpy );
+						sinon.assert.calledOnce( onShowSpy );
+					}, { priority: 'lowest' } );
+
+					dialogPlugin.fire( 'show', { onShow: onShowSpy } );
+				} );
+			} );
+
+			describe( 'hide event listeners', () => {
+				it( 'executing `_hide()` method ', () => {
+					const spy = sinon.spy( dialogPlugin, '_hide' );
+
+					dialogPlugin.fire( 'hide' );
+
+					sinon.assert.calledOnce( spy );
+				} );
+
+				it( 'executing `onHide` callback', () => {
+					const spy = sinon.spy( () => {} );
+
+					dialogPlugin._onHide = spy;
+
+					dialogPlugin.fire( 'hide' );
+
+					sinon.assert.calledOnce( spy );
+				} );
+
+				it( 'should execute `_hide()` before `_onHide`', () => {
+					const _hideSpy = sinon.spy( dialogPlugin, '_hide' );
+					const onHideSpy = sinon.spy( () => {} );
+
+					dialogPlugin._onHide = onHideSpy;
+
+					dialogPlugin.on( 'hide', () => {
+						sinon.assert.notCalled( _hideSpy );
+						sinon.assert.notCalled( onHideSpy );
+					}, { priority: 'high' } );
+
+					dialogPlugin.on( 'hide', () => {
+						sinon.assert.calledOnce( _hideSpy );
+						sinon.assert.notCalled( onHideSpy );
+					}, { priority: -1 } );
+
+					dialogPlugin.on( 'hide', () => {
+						sinon.assert.calledOnce( _hideSpy );
+						sinon.assert.calledOnce( onHideSpy );
+					}, { priority: 'lowest' } );
+
+					dialogPlugin.fire( 'hide' );
+				} );
+			} );
+
+			describe( 'multiroot integration', () => {
+				it( 'should update the dialog view position whenever a root state is changed', () => {
```

Comment:
We're missing a test that verifies that this does not happen upon root attribute change.